__main__.py

```diff
@@ -6,16 +6,33 @@
     args = [x for x in sys.argv[1:] if not x.startswith("-")]
     flags = [x for x in sys.argv[1:] if x.startswith("-")]
 
-    fn = args[0]
+    filename = args[0]
     start_time = time.time()
+    
+    with_progress = "--progress" in flags
+    json_output = "--json" in flags
+    only_header = "--header-only" in flags
+    validate_data_only = "--data-only" in flags
+    
+    
+    # Sanity check: can't use both at once
+    if only_header and validate_data_only:
+        print("Cannot use both --header-only and --data-only at the same time", file=sys.stderr)
+        sys.exit(2)
 
     try:
-        parse(filename=fn, with_progress="--progress" in flags, with_tree=False)
-        if "--json" not in flags:
+        parse(
```

Comment:
Sorry, but maybe also rewrite to argparse and just pass the argparse.Namespace object. I know this is 100% on me, but this many keyword args make me a bit nervous.